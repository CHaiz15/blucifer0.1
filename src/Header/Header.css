.header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  height: 8vh;
  background-color: #3C5576;
  box-shadow: 0 2px 10px -2px rgba(0,0,0,.8);
  z-index: 100;
  overflow: hidden;
  position: fixed;
  top: 0;
  width: 100%;
  color: #fff;
  text-shadow: 0 0 2px #000;
  padding: 0 2% 0 2%;
}

.logout-btn {
  padding: 5%;
  background-color: transparent;
  border: none;
  color: #fff;
  text-shadow: 0 0 2px #000;
  font-size: 1.2em;
  font-weight: bold;
}

