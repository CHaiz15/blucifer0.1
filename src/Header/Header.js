import React from 'react';
import './Header.css';
import logo from '../assets/DarkLogo.png'
import { Link } from 'react-router-dom';

const logout = () => {
    window.location.href = '/';
}

const Header = ({ name, purpose, totalFavorites }) => {
  return (
    <header className="header">
      <div className='welcome-holder'>
        <img className='blucifer-logo' src={logo} alt='logo'/>
        <h3 className='user-welcome'>Welcome, {name}!</h3>
      </div>
      <Link to='/nav/areas' className="logout-btn">{purpose}, made easy.</Link>
      <Link to='/nav/favorites' className="logout-btn">Favorites({totalFavorites})</Link>
      <button className="logout-btn" onClick={logout}>Logout</button>
    </header>
  )
}

export default Header;
