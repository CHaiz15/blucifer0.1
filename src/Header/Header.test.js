import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';
import { shallow } from 'enzyme'

describe('Header', () => {
  it('should be an instance of Header component', () => {
    const wrapper = shallow(<Header/>);
      expect(wrapper).toMatchSnapshot();
  });
})