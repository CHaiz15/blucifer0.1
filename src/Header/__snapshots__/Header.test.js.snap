// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be an instance of Header component 1`] = `
<header
  className="header"
>
  <div
    className="welcome-holder"
  >
    <img
      alt="logo"
      className="blucifer-logo"
      src="DarkLogo.png"
    />
    <h3
      className="user-welcome"
    >
      Welcome, 
      !
    </h3>
  </div>
  <h3>
    , made easy.
  </h3>
  <button
    className="logout-btn"
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
